src/Analyzers/CSharp/CodeFixes/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.cs

```diff
@@ -82,10 +82,27 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 }
 
                 var switchStatement = (SwitchStatementSyntax)switchLocation.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+                // If the swith statement expression is being implicitly converted then we need to explicitly cast the expression
+                // before rewriting as a switch expression
+                var expressionISymbolType = semanticModel.GetSymbolInfo(switchStatement.Expression).Symbol.GetSymbolType();
+                var expConversionISymbolType = semanticModel.GetTypeInfo(switchStatement.Expression).ConvertedType;
+
+                var needsCast = false;
+                SwitchStatementSyntax newSwitch = null;
+                if (expConversionISymbolType != expressionISymbolType && expConversionISymbolType != null)
+                {
+                    needsCast = true;
+                    newSwitch = switchStatement.Update(switchStatement.SwitchKeyword, switchStatement.OpenParenToken,
+                        switchStatement.Expression.Cast(expConversionISymbolType).WithAdditionalAnnotations(Formatter.Annotation),
+                        switchStatement.CloseParenToken, switchStatement.OpenBraceToken,
+                        switchStatement.Sections, switchStatement.CloseBraceToken);
```

Comment:
i don't liek that you're changing the switch-statement itself (the original guy).  I woudl expect that we'd simply add the cast when creating the switch-expression.

Reply:
I actually think you need to move this down to the re-writer in `ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs`. The syntax editor will throw if you give it a modified node. As Cyrus said earlier. I would just change to rewriter to always add the cast and then add the `Simplifier.Annotation`  like so:

```csharp
editor.ReplaceNode(switchStatement, switchExpression.WithAdditionalAnnotations(Formatter.Annotation, Simplifier.Annotation));
```

Reply:
Note: i believe `.Cast` will autoamtically add teh simplifier annotation.  :)

Reply:
The changes have been made. The cast now occurs every time and is removed by the Simplifier if not necessary. The change also occurs in the Rewriter (when the switch-expression is made) :)